crates/bevy_ui/src/focus.rs

```diff
@@ -1,25 +1,23 @@
-use crate::{
-    CalculatedClip, ComputedNode, DefaultUiCamera, ResolvedBorderRadius, TargetCamera, UiScale,
-    UiStack,
-};
+use crate::{CalculatedClip, ComputedNode};
+#[cfg(feature = "bevy_render")]
+use crate::{DefaultUiCamera, ResolvedBorderRadius, TargetCamera, UiScale, UiStack};
+#[cfg(feature = "bevy_render")]
 use bevy_ecs::{
-    change_detection::DetectChangesMut,
-    entity::Entity,
-    prelude::{Component, With},
-    query::QueryData,
-    reflect::ReflectComponent,
+    change_detection::DetectChangesMut as _,
+    prelude::With,
     system::{Local, Query, Res},
 };
```

Comment:
Nit: I personally prefer keeping feature gates to an absolute minimum, so I would write a larger `use` block for the whole gate instead:

```rust
#[cfg(feature = "bevy_render")]
use {
    crate::{DefaultUiCamera, ResolvedBorderRadius, TargetCamera, UiScale, UiStack},
    bevy_ecs::{...},
    ...
};
```

Fewer lines, and I think it's easier to read. "_For the `bevy_render feature, we need **these** items_"

Reply:
I wasn't aware that the formatter would allow that. I assumed it would split it by crate.